feat: add index parameter to customAnimation

3000-2 opened the pull request:
### Description

This PR updates the customAnimation function signature to include an index parameter, allowing users to apply custom animations based on the item’s index.

### Example

https://github.com/user-attachments/assets/388f5f4b-84d7-4aec-ba25-a9069ff06fb9

### Review

- [x] I self-reviewed this PR

dohooo commented:
Could you resolve the conflicts and add a simple usage case for this change? For instance, a new demo to illustrate the scenario where this argument is needed. I would greatly appreciate it!

dohooo commented:
released https://github.com/dohooo/react-native-reanimated-carousel/pull/736

3000-2 commented:
> ~Could you resolve the conflicts and add a simple usage case for this change? For instance, a new demo to illustrate the scenario where this argument is needed. I would greatly appreciate it!~
> 
> I've addressed these and added tests in another PR. I'll add credits for you, thanks! #735

Sorry for the late response, and thank you for taking care of this!
